### PR TITLE
Integrating Google Authentication to Sign Up Form

### PR DESCRIPTION
**Pull request by George-Sucuzhanay**
Testing Instructions:

1. run both client and server with npm run dev
2. click on the sign in button
3. click on sign in with google with your google account
4. you should be redirected to http://localhost:3000/home 
5. you should see Hi, "user name" on the upper right hand corner

### CODE REVIEW

**Comment by George-Sucuzhanay** [comment]
will be working on resolve merge conflict by testing locally, do not review just yet.

File: server/routes/user.js [hunk 1]
```diff
@@ -166,4 +167,55 @@ router.post('/login', async (req, res) => {
...
+router.post('/users/auth/google', async (req, res) => {
+    console.log("this post api endpoint was hit")
+    const { email } = req.body;
+    console.log("this post call found:", email)
+    try {
+      let user = await User.findOne({ username: email });
+      console.log("this is a user found during post call: ", user)
+
+      if (!user) {
+        console.log("no user was found :(")
```
Comment (EricLi-Dev) [inline server/routes/user.js:196]: .

File: server/routes/user.js
Comment (EricLi-Dev) [inline server/routes/user.js:1]: Can we add unit tests for these functions w/ our jest setup?

File: server/server.js [hunk 2]
```diff
@@ -2,6 +2,8 @@
 const express = require("express");
 const cors = require('cors')
 const app = express();
+require('dotenv').config();
```
Comment (EricLi-Dev) [inline server/server.js:5]: Do we use this?

**Review by EricLi-Dev — CHANGES_REQUESTED** [review]
I see a lot of console.log()'s that I think should be removed first before merging, but the overall logic of the authentication looks good to me. Some tests on your backend functions would be good practice for our repo! Let me know what you think.

**Review by Dulinag — COMMENTED** [review]
I do not see changes reflected on the front end with Google auth. Additionally, the Google auth is supposed to get the username and display it with the user modal. A simple successful console log isn't enough as this is already a 2-month-old ticket. If it were 2 different tickets at first it would be fine but the period of time reflected on this ticket is plentiful for major changes on the front end with our login component.

**Comment by Dulinag** [comment]
Make the username displayed on the front end as soon as possible as more components need attention rather than Google auth. Also, uniformity is needed, Please do not use different styles such as blocky big images. Keep everything in the modal which the user can click on and see information.

**Review by Dulinag — CHANGES_REQUESTED** [review]
make changes soon. Thanks

File: server/routes/user.js
Comment (George-Sucuzhanay) [reply server/routes/user.js:1]: for purposes of this ticket we can discuss further but for now its not a priority

File: server/server.js (hunk 2 above)
Comment (George-Sucuzhanay) [reply server/server.js:5]: yes, there is an env that i am using in the server

File: client/components/login.jsx [hunk 3]
```diff
@@ -36,6 +54,34 @@ function Login() {
...
+            Accept: 'application/json'
+          }
+        });
+
+        
+        
+        user.profile = res.data; // Include profile information in the user object
+        console.log(user)
+        signIn(user);  // Update global user state
+        setModalOpen(false);  // Close the modal upon successful login
```
[the remark this answers is not in the document]
Comment (George-Sucuzhanay) [reply client/components/login.jsx]: after the user successfully logs in the modal need to close.

**Review by EricLi-Dev — APPROVED** [review]
lgtm

**Review by Dulinag — APPROVED** [review]
Lgtm

**Pull request merged**